apps/mivisionx_validation_tool/rali_setup.py

```diff
@@ -1,422 +1,511 @@
-		self.reset()
-		
-	def get_next_augmentation(self):
-                if self.raliIsEmpty() == 1:
-			#raise StopIteration
-			return -1
-		self.renew_parameters()
-		if self.run() != 0:
-			#raise StopIteration
-			return -1
-		self.copyToNPArray(self.out_image)
-		if(TensorLayout.NCHW == self.tensor_format):
-			self.copyToTensorNCHW(self.out_tensor, self.multiplier, self.offset, self.reverse_channels, self.tensor_dtype)
-
-		else:
-			self.copyToTensorNHWC(self.out_tensor, self.multiplier, self.offset, self.reverse_channels, self.tensor_dtype)
-
-		return self.out_image , self.out_tensor
-
-	def get_rali_list(self, raliMode, model_batch_size):
-		if model_batch_size == 16:
-			if raliMode == 1:
-				self.rali_list = raliList_mode1_16
-			elif raliMode == 2:
-				self.rali_list = raliList_mode2_16
-			elif raliMode == 3:
-				self.rali_list = raliList_mode3_16
-			elif raliMode == 4:
-				self.rali_list = raliList_mode4_16
-			elif raliMode == 5:
-				self.rali_list = raliList_mode5_16
-		elif model_batch_size == 64:
-			if raliMode == 1:
-				self.rali_list = raliList_mode1_64
-			elif raliMode == 2:
-				self.rali_list = raliList_mode2_64
-			elif raliMode == 3:
-				self.rali_list = raliList_mode3_64
-			elif raliMode == 4:
-				self.rali_list = raliList_mode4_64
-			elif raliMode == 5:
-				self.rali_list = raliList_mode5_64
-				
-		return self.rali_list
+# Class to initialize Rali and call the augmentations 
+class InferencePipe(Pipeline):
+    def __init__(self, image_validation, model_batch_size, raliMode, c_img, h_img, w_img, rali_batch_size, tensor_dtype, multiplier, offset, tensor_layout, num_threads, device_id, data_dir, crop, rali_cpu = True):
+        super(InferencePipe, self).__init__(rali_batch_size, num_threads, device_id, seed=12 + device_id,rali_cpu=rali_cpu)
+        #world_size = 1
+        #local_rank = 0
+        rali_device = 'cpu' if rali_cpu else 'gpu'
+        decoder_device = 'cpu' if rali_cpu else 'mixed'
+        device_memory_padding = 211025920 if decoder_device == 'mixed' else 0
+        host_memory_padding = 140544512 if decoder_device == 'mixed' else 0
+        
+        self.validation_dict = {}
+        self.process_validation(image_validation)
+        self.set_seed(0)
+        self.aug_strength = 0
+        self.model_batch_size = model_batch_size
+        self.raliMode =  raliMode
+        self.data_dir = data_dir
+        self.c_img = c_img
+        self.h_img = h_img
+        self.w_img = w_img
+        self.rali_batch_size = rali_batch_size
+        self.tensor_dtype = tensor_dtype
+        self.multiplier = multiplier
+        self.offset = offset
+        self.tensor_layout = tensor_layout
+        self.reverse_channels = False
+        #for tensor output
+        self.bs = self.rali_batch_size
+        if self.tensor_dtype == types.FLOAT:
+            self.out_tensor = np.zeros(( self.bs*self.model_batch_size, self.c_img, int(self.h_img/self.bs), self.w_img,), dtype = "float32")
```

Comment:
This has to be changed according to the latest changes.
We create empty torch tensors instead of numpy arrays
Please refer [](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/b7e3bcaa55098522cdcee8d3fce172f2d77a3413/rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py#L71)

Reply:
@swetha097 Can you add the link for me to refer?